### DISCUSSION TITLE
grid columns based on screen size

### DISCUSSION THREAD

**Discussion by sasgosh** [quasarframework/quasar#10104] [General - Components / Directives / etc] [answered]
Hello, 

I want to make grid where I have 1 column when screen is xs, 2 columns when sm and md and 3 columns for larger screens (without 4th column). I managed to do it somehow but I don't know if there is an easier way to do that?

I have 3 classes:

```
.columns1 {display: grid; grid-template-columns: repeat(1, 1fr);}
.columns2 {display: grid; grid-template-columns: repeat(2, 1fr);}
.columns3 {display: grid; grid-template-columns: repeat(3, 1fr);}
```

And elements (which are always different count), for example:

```
<div :class="{'columns1': $q.screen.xs, 'columns2': $q.screen.sm || $q.screen.md, 'columns3': $q.screen.gt.md}">
    <div class="q-ma-sm">box</div>
    <div class="q-ma-sm">box</div>
    <div class="q-ma-sm">box</div>
    <div class="q-ma-sm">box</div>
    <div class="q-ma-sm">box</div>
</div>
```

Is there a better way to do that?

I made codepen: https://codepen.io/SaleCar/pen/jOmmJZx

**Comment by MilosPaunovic** [comment, marked as answer]
Hello @SaleCar,

Have you tried with the Quasar classes, like this:
```js
<div class="row">
   <div class="col col-xs-12 col-sm-6 col-md-4">Some content</div>
</div>
```

This should give you single column on `xs` screens, two colums on `sm` ones and three for `md` breakpoint and above.

**Reply by sasgosh** [reply]
Yes, I tried everything including that. And it didn't work as I expected. But now I see what the problem is. Elemtents separation. The margin expanded the elements and then they break badly. When I changed q-mr-sm into q-pr-sm, everything works properly! 

I also tried to use q-col-gutter for separation but I found that it doesn't suit me either, it doesn't separate the element from the right edge. When we have only one column (on mobile phones) then the element's right edge is glued to the right edge of the screen. I think it would be best for Gutter to apply the margins equally on all sides and not just left and top. 

Anyway, the problem is solved. This yours + padding for separation. Thanks.
